src/__tests__/hooks.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { render } from "@testing-library/react";
+import React, { Dispatch } from "react";
+
+import { Auth0ProviderAction } from "../actions";
+import * as hooks from "../hooks";
+import { useAuth0, useLog } from "../hooks";
+import { Auth0ProviderContext, AuthenticatedState } from "../types";
+
+const TestComponent: React.FC<{ debug?: boolean; logMessage?: string }> = ({ debug, logMessage }) => {
+  const log = useLog(debug);
+  log(logMessage || "Logging!");
+  return <div />;
+};
+
+describe("hooks", () => {
+  describe("useLog", () => {
+    let origConsole: Pick<typeof window.console, "log" | "warn" | "error">;
+
+    beforeEach(() => {
+      origConsole = {
+        log: window.console.log,
+        warn: window.console.warn,
+        error: window.console.error,
+      };
+
+      window.console.log = jest.fn();
+      window.console.warn = jest.fn();
+      window.console.error = jest.fn();
+    });
+
+    afterEach(() => {
+      window.console.log = origConsole.log;
+      window.console.warn = origConsole.warn;
+      window.console.error = origConsole.error;
+    });
+
+    it("Does not log anything if no arguments are provided", () => {
+      render(<TestComponent />);
+      expect(window.console.log).not.toHaveBeenCalled();
+    });
+
+    it("Does not log anything if the debug flag is set to false", () => {
+      render(<TestComponent debug={false} />);
+      expect(window.console.log).not.toHaveBeenCalled();
+    });
+
+    it("Logs if the debug flag is set to true", () => {
+      const logString = "test";
+      render(<TestComponent debug={true} logMessage={logString} />);
+      expect(window.console.log).toBeCalledWith(logString);
+    });
+  });
+
+  describe("useAuth0", () => {
+    it("throws an error if the auth0 context is not present", async () => {
+      const FakeFC: React.FC = () => {
+        const auth0 = useAuth0();
+
+        return <div data-testid="username">{auth0.user?.name}</div>;
+      };
+
+      expect(() => render(<FakeFC />)).toThrowErrorMatchingSnapshot();
```

Comment:
maybe use `toThrowErrorMatchingInlineSnapshot`
```
expect(() => render(<FakeFC />)).toThrowErrorMatchingInlineSnapshot();
```

it will move the snapshot contents to here something like
```
expect(() => render(<FakeFC />)).toThrowErrorMatchingInlineSnapshot(`"useAuth0 may only be used within the context of an <Auth0Provider />"`);
```

which is a whole lot easier to understand than having to reference the other file.  you still get all the automatic updates with `-u` that you would want.  The other benefit of the `InlineSnapshot` variants is that it forces the snapshots to remain small

Reply:
done